content/courses/advanced-ada/chapters/packages.rst

```diff
@@ -181,16 +181,197 @@ possible to access the information from a private package in the private part
 of a non-private child package.
 
 
-Partial dependencies
---------------------
+Limited Visibility
+------------------
 
-.. admonition:: Relevant topics
+Sometimes, we might face the situation where two packages depend on
+information from each other. Let's consider a package :ada:`A` that depends
+on a package :ada:`B`, and vice-versa:
 
-    - `Context Clauses - With Clauses <http://www.ada-auth.org/standards/2xrm/html/RM-10-1-2.html>`_
+.. code:: ada compile_button project=Courses.Advanced_Ada.Packages.Circular_Dependency
+    :class: ada-expect-compile-error
 
-.. todo::
+    with B; use B;
 
-    Complete section!
+    package A is
+
+       type T1 is record
+          Value : T2;
+       end record;
+
+    end A;
+
+    with A; use A;
+
+    package B is
+
+       type T2 is record
+          Value : T1;
+       end record;
+
+    end B;
+
+Here, we have two
+:ref:`mutually dependent types <Adv_Ada_Mutually_Dependent_Types>` :ada:`T1`
+and :ada:`T2`, which are declared in two packages :ada:`A` and :ada:`B` that
+refer to each other. These with clauses constitute a circular dependency, so
+the compiler cannot compile any of those packages.
```

Comment:
Consider changing "any" to "either" because "any" suggests several, whereas there are precisely two in this scenario...

Reply:
Thanks, I've just corrected it.